### PR TITLE
Ensure any .info on Masked instances is propagated correctly.

### PR DESCRIPTION
**Pull request by mhvk**
I thought I had dealt with `.info` in the original implementation, but I clearly missed updating it in viewing or slicing. With this change, ``MaskedQuantity`` can now be correctly written to, e.g., ECSV format with ``serialize_method='data_mask'``. So, added tests for serialization of masked quantities. It should also fix the one remaining issue in using `Masked` inside `Time`.

### Description

This pull request is to address ...

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will
review this pull request of some common things to look for. This list
is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] If the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/utils/data_info.py
Comment (mhvk) [inline astropy/utils/data_info.py:360]: I'm not sure how we have avoided to run into this before, but without this one cannot transfer info from a `Quantity` to a `MaskedQuantity`, since the former does not actually have `serialize_methods`.

File: astropy/utils/masked/core.py
Comment (mhvk) [inline astropy/utils/masked/core.py:354]: So, the answer to this `TODO` was to ensure `info` was propagated correctly, which means 'serialize_method' is passed on as well. The system actually worked brilliantly once I realized what was wrong.

**Comment by mhvk** [comment]
@taldcroft - sorry for the many requests for review! I tried to split into small pieces but it seems there is more dependence between those than I had realized. In any case, this is the first one!

File: astropy/utils/data_info.py
Comment (mhvk) [reply astropy/utils/data_info.py:360]: To clarify the relation to #11917: without this change one cannot do `masked_column.info = column.info`.

**Comment by mhvk** [comment]
@taldcroft - if you have some time, this one is holding up making `QTable(..., masked=True)` work...

**Review by taldcroft — APPROVED** [review]
This looks good. I do see some exact replication of changes in #11917 (which is ready for merge apart from a changelog), so I'm still not 100% clear on the path here. Once one of the pair are merged the other will need a rebase and pass tests again.

**Comment by eteq** [comment]
(this ended up in v4.3.1 due to some unintended early backporting)

**Pull request merged**
